extensions/Worker.Extensions.Shared/Reflection/ParameterBinder.cs

```diff
@@ -94,9 +100,8 @@ await foreach (object item in pageable)
         private static bool IsListInterface(Type type)
         {
             return type.IsGenericType &&
-                (type.GetGenericTypeDefinition() == typeof(IList<>)
-                || type.GetGenericTypeDefinition() == typeof(ICollection<>)
-                || type.GetGenericTypeDefinition() == typeof(IEnumerable<>));
+                (validListInterfaceTypes.Contains(type.GetGenericTypeDefinition()) ||
+                 type.GetInterfaces().Select(t => t.GetGenericTypeDefinition()).Intersect(validListInterfaceTypes).Any());
```

Comment:
Checking all interfaces is too broad, there is no guarantee we can instantiate the declared interface. For example, I could add this type:

``` CSharp
public interface ICustomList<T> : IList<T> { }
```

This would pass the `IsListInterface` check, but we cannot actually satisfy that type with `List<T>` - or any other concrete type we know of.

I think the best option will be to explicitly add support for `IReadOnlyList<T>`, and also `IReadOnlyCollection<T>` as `List<T>` implements it as well.

Reply:
I get what you're saying as I read the intent of the method differently than how you did.  After reading your comment and re-reading the code I get that the intent is to identify if the Type is a generic interface on List<T>.

I don't think a static list of interfaces is the right solution though because if .NET changes the set of interfaces that List<T> supports and someone tries to leverage one of the new interfaces then they'll be blocked until you rev this shared code.

I think a better fix would be to check the passed in Type against the generic interfaces of List<> to see if it's one of them.  Then as List<T>'s interfaces change you don't need to rev this shared code.  I'll see if I can code that up later tonight.

Reply:
My latest commit changes the IsListInterface to check against every generic interface on `List<T>` to see if the passed in type is one on `List<T>`.  I've also added a test that will verify that every generic interface will return a list as expected.

I ended up using a little more LINQ then I intended to though and I understand this has a performance hit.  I didn't spot any perf tests so I don't know the exact impact.  I expect it to be small but I'm sure that even small increases are a problem for something this widely used.

If you'd rather not take any performance hit I can go back to the static list though I'd keep the test I added to validate that all of the interfaces are supported.  That way if `List<T>` gets updated in the future it will be easy to figure out that more interfaces need to be added.